lib/ansible/modules/cloud/amazon/dynamodb.py

```diff
@@ -0,0 +1,390 @@
+                  "ExpressionAttributeNames": expression_attribute_names}
+
+    else:
+        kwargs = {"TableName": table,
+                  "FilterExpression": filter_expression,
+                  "ExpressionAttributeValues": expression_attribute_values}
+
+    response = get_with_backoff(connection, **kwargs)
+
+    changed = False
+
+    response['returned_items'] = response.pop('Items')
+
+    return response, changed
+
+
+def put(connection, table, item):
+
+    response = connection.put_item(
+        TableName=table,
+        Item=item,
+    )
+
+    changed = True
+    return response, changed
+
+
+def update(connection, table, primary_key, update_expression,
+           condition_expression, expression_attribute_names,
+           expression_attribute_values):
+
+    if condition_expression:
+        response = connection.update_item(TableName=table, Key=primary_key, UpdateExpression=update_expression,
+                                          ConditionExpression=condition_expression, ExpressionAttributeValues=expression_attribute_values)
+    elif expression_attribute_names:
+        response = connection.update_item(TableName=table, Key=primary_key, UpdateExpression=update_expression,
+                                          ExpressionAttributeValues=expression_attribute_values, ExpressionAttributeNames=expression_attribute_names)
+    elif update_expression and expression_attribute_values:
+        response = connection.update_item(TableName=table, Key=primary_key, UpdateExpression=update_expression,
+                                          ExpressionAttributeValues=expression_attribute_values)
+    else:
+        response = connection.update_item(
+            TableName=table, Key=primary_key, UpdateExpression=update_expression)
+
+    changed = True
+    return response, changed
+
+
+def delete(connection, table, primary_key, condition_expression,
+           expression_attribute_values):
+
+    response = connection.delete_item(TableName=table, Key=primary_key, ConditionExpression=condition_expression,
+                                      ExpressionAttributeValues=expression_attribute_values)
+
+    changed = True
+    return response, changed
+
+
+def main():
+    argument_spec = dict(
+        table=dict(required=True),
+        action=dict(required=True, choices=["get", "put", "update", "delete"]),
+        primary_key=dict(required=False, type='dict'),
+        filter_expression=dict(required=False, type='str'),
+        item=dict(required=False, type='dict'),
+        condition_expression=dict(required=False, type='str'),
+        expression_attribute_names=dict(required=False, type='dict'),
+        expression_attribute_values=dict(required=False, type='dict'),
+        update_expression=dict(required=False, type='str')
+    )
+
+    module = AnsibleAWSModule(argument_spec=argument_spec,
+                              supports_check_mode=False)
```

Comment:
is there a reason this can't support check mode?

Reply:
ill see if i can support it